src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServer1.java

```diff
@@ -0,0 +1,149 @@
+import com.google.cloud.tools.appengine.api.devserver.RunConfiguration;
+import com.google.cloud.tools.appengine.api.devserver.StopConfiguration;
+import com.google.cloud.tools.appengine.cloudsdk.internal.args.DevAppServerArgs;
+import com.google.cloud.tools.appengine.cloudsdk.internal.process.ProcessRunnerException;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.Maps;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Classic Java SDK based implementation of {@link AppEngineDevServer}.
+ */
+public class CloudSdkAppEngineDevServer1 implements AppEngineDevServer {
+
+  private final CloudSdk sdk;
+
+  private static final String DEFAULT_HOST = "localhost";
+  private static final int DEFAULT_PORT = 8080;
+
+  public CloudSdkAppEngineDevServer1(CloudSdk sdk) {
+    this.sdk = Preconditions.checkNotNull(sdk);
+  }
+
+  /**
+   * Starts the local development server, synchronously or asynchronously.
+   *
+   * @throws AppEngineException I/O error in the Java dev server.
+   */
+  @Override
+  public void run(RunConfiguration config) throws AppEngineException {
+    Preconditions.checkNotNull(config);
+    Preconditions.checkNotNull(config.getServiceDirectories());
+    Preconditions.checkArgument(config.getServiceDirectories().size() > 0);
+    AppEngineDescriptor appengineWeb;
+    try {
+      // TODO(ludo: Make sure we support the case when more than 1 service is given...
+      FileInputStream is = new FileInputStream(
+              new File(config.getServiceDirectories().get(0), "WEB-INF/appengine-web.xml"));
+      appengineWeb = AppEngineDescriptor.parse(is);
+    } catch (IOException e) {
+      throw new AppEngineException(e);
+    }
+    List<String> arguments = new ArrayList<>();
+
+    Map<String, String> env = Maps.newHashMap();
+    if (!Strings.isNullOrEmpty(config.getJavaHomeDir())) {
+      env.put("JAVA_HOME", config.getJavaHomeDir());
+    }
+
+    List<String> jvmArguments = new ArrayList<>();
+    boolean isJava8 = (appengineWeb.getRuntime() != null)
+            && appengineWeb.getRuntime().startsWith("java8");
+    if (isJava8) {
+      jvmArguments.add("-Duse_jetty9_runtime=true");
+      jvmArguments.add("-D--enable_all_permissions=true");
+    } else {
+      // Add in the appengine agent
+      String appengineAgentJar = new File(
+              sdk.getJavaAppEngineSdkPath().toFile(), "agent/appengine-agent.jar")
+              .getAbsolutePath();
+      jvmArguments.add("-javaagent:" + appengineAgentJar);
+    } 
+    arguments.addAll(DevAppServerArgs.get("server", config.getHost()));
+    arguments.addAll(DevAppServerArgs.get("address", config.getPort()));
+    arguments.addAll(DevAppServerArgs.get("jvm_flag", config.getJvmFlags()));
```

Comment:
automatic restart in devappserver2 restarts the server on every file change.  It was a surprise to us too. We had to turn it off for debug and code sessions to work. 

if log control isn't available we can leave it out for now.

running dev_appserver.sh on the old SDK I do see a disable_update_check flag available.